discord/state.py

```diff
@@ -165,9 +167,9 @@ class ConnectionState:
     def __init__(
         self,
         *,
-        dispatch: Callable,
-        handlers: Dict[str, Callable],
-        hooks: Dict[str, Callable],
+        dispatch: Callable[..., Any],
+        handlers: Dict[str, Callable[..., Any]],
+        hooks: Dict[str, Callable[..., Any]],
```

Comment:
Same as in `client.py` these are coroutines.
```suggestion
        hooks: Dict[str, Callable[..., Coroutine[Any, Any, Any]]],
```